commons/src/main/java/org/codice/ddf/admin/commons/requests/RequestUtils.java

```diff
@@ -0,0 +1,232 @@
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.ssl.SSLContexts;
+import org.apache.http.util.EntityUtils;
+import org.codice.ddf.admin.api.handler.ConfigurationMessage;
+import org.codice.ddf.admin.api.handler.MessageBuilder;
+import org.codice.ddf.admin.api.handler.report.ProbeReport;
+import org.codice.ddf.admin.api.handler.report.Report;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.net.HttpHeaders;
+
+public class RequestUtils {
+
+    //Probe return types
+    public static final String CONTENT_TYPE = "contentType";
+    public static final String CONTENT = "content";
+    public static final String STATUS_CODE = "statusCode";
+
+    //Success types
+    public static final String EXECUTED_REQUEST = "EXECUTED_REQUEST";
+    public static final String CONNECTED = "CONNECTED";
+    private static final Map<String, String> SUCCESS_DESCRIPTIONS = ImmutableMap.of(
+            CONNECTED, "Successfully established a connection with the endpoint.",
+            EXECUTED_REQUEST, "A request was successfully sent and a response was received from the endpoint.");
+
+    //Failure types
+    public static final String CERT_ERROR = "CERT_ERROR";
+    public static final String CANNOT_CONNECT = "CANNOT_CONNECT";
+    private static final Map<String, String> FAILURE_DESCRIPTIONS = ImmutableMap.of(
+            CANNOT_CONNECT, "The URL provided could not be reached.",
+            CERT_ERROR, "The discovered source has incorrectly configured SSL certificates and is insecure.");
+
+    //Warning types
+    public static final String UNTRUSTED_CA = "UNTRUSTED_CA";
+    public static final Map<String, String> WARNING_DESCRIPTIONS = ImmutableMap.of(UNTRUSTED_CA,
+            "The discovered URL has incorrectly configured SSL certificates and is likely insecure.");
+
+    private static final MessageBuilder REQUEST_MESSAGE_BUILDER = new MessageBuilder(
+            SUCCESS_DESCRIPTIONS,
+            FAILURE_DESCRIPTIONS,
+            WARNING_DESCRIPTIONS);
+
+    public static final int PING_TIMEOUT = 500;
+
+    /**
+     * Sends a get request to the specified url. It does NOT check the response status code or body.
+     * SUCCESS TYPES - EXECUTED_REQUEST
+     * WARNING TYPES - UNTRUSTED_CA
+     * FAILURE TYPES - CANNOT_CONNECT, CERT_ERROR
+     * RETURN TYPES -  CONTENT_TYPE, CONTENT, STATUS_CODE
+     *
+     * @param url
+     * @param userName - username to add to basic auth
+     * @param password - user password to add to basic auth
+     * @return report
+     */
+    public static ProbeReport sendGetRequest(String url, String userName, String password) {
+        Report connectReport = endpointIsReachable(url);
+        if(connectReport.containsFailureMessages()) {
+            return new ProbeReport(connectReport.messages());
+        }
+        HttpGet request = new HttpGet(url);
+
+        if (url.startsWith("https") && userName != null && password != null) {
```

Comment:
Pro - slightly less code duplication.
Con - pass the username and password into yet another method.
I'm indifferent.